lib/sda/node.go

```diff
@@ -313,8 +333,44 @@ func (n *Node) DispatchChannel(msgSlice []*SDAData) error {
 	return nil
 }
 
-// DispatchMsg will dispatch this SDAData to the right instance
-func (n *Node) DispatchMsg(sdaMsg *SDAData) error {
+// DispatchMsg takes a message and puts it into a queue for later processing.
+// This allows a protocol to have a backlog of messages.
+func (n *Node) DispatchMsg(msg *SDAData) {
+	dbg.Lvl3(n.Myself(), "Received message")
+	n.msgDispatchQueueMutex.Lock()
+	n.msgDispatchQueue = append(n.msgDispatchQueue, msg)
+	dbg.Lvl3(n.Myself(), "DispatchQueue-length is", len(n.msgDispatchQueue))
+	if len(n.msgDispatchQueue) == 1 {
+		n.msgDispatchQueueWait <- true
+	}
+	n.msgDispatchQueueMutex.Unlock()
+}
+
+func (n *Node) dispatchMsgReader() {
+	for {
+		n.msgDispatchQueueMutex.Lock()
+		if len(n.msgDispatchQueue) > 0 {
+			dbg.Lvl3(n.Myself(), "Read message and dispatching it",
+				len(n.msgDispatchQueue))
+			msg := n.msgDispatchQueue[0]
+			n.msgDispatchQueue = n.msgDispatchQueue[1:]
+			n.msgDispatchQueueMutex.Unlock()
+			n.dispatchMsgToProtocol(msg)
```

Comment:
Yes, at least display it. Otherwise, if something behaves not as expected, we have to search and find this line I guess (and this might take some time). There is even an open issue related to this: #160